tests/pmdarima/test_pmdarima_model_export.py

```diff
@@ -0,0 +1,427 @@
+        mlflow.pmdarima.log_model(
+            auto_arima_object_model, "model", pip_requirements=[f"-c {req_file.strpath}", "b"]
+        )
+        _assert_pip_requirements(
+            mlflow.get_artifact_uri("model"),
+            ["mlflow", "b", "-c constraints.txt"],
+            ["a"],
+            strict=True,
+        )
+
+
+def test_pmdarima_log_model_with_extra_pip_requirements(auto_arima_model, tmpdir):
+    default_reqs = mlflow.pmdarima.get_default_pip_requirements()
+
+    # Path to a requirements file
+    req_file = tmpdir.join("requirements.txt")
+    req_file.write("a")
+    with mlflow.start_run():
+        mlflow.pmdarima.log_model(
+            auto_arima_model, "model", extra_pip_requirements=req_file.strpath
+        )
+        _assert_pip_requirements(mlflow.get_artifact_uri("model"), ["mlflow", *default_reqs, "a"])
+
+    # List of requirements
+    with mlflow.start_run():
+        mlflow.pmdarima.log_model(
+            auto_arima_model, "model", extra_pip_requirements=[f"-r {req_file.strpath}", "b"]
+        )
+        _assert_pip_requirements(
+            mlflow.get_artifact_uri("model"), ["mlflow", *default_reqs, "a", "b"]
+        )
+
+    # Constraints file
+    with mlflow.start_run():
+        mlflow.pmdarima.log_model(
+            auto_arima_model, "model", extra_pip_requirements=[f"-c {req_file.strpath}", "b"]
+        )
+        _assert_pip_requirements(
+            model_uri=mlflow.get_artifact_uri("model"),
+            requirements=["mlflow", *default_reqs, "b", "-c constraints.txt"],
+            constraints=["a"],
+            strict=False,
+        )
+
+
+def test_pmdarima_model_save_without_conda_env_uses_default_env_with_expected_dependencies(
+    auto_arima_model, model_path
+):
+    mlflow.pmdarima.save_model(auto_arima_model, model_path)
+    _assert_pip_requirements(model_path, mlflow.pmdarima.get_default_pip_requirements())
+
+
+def test_pmdarima_model_log_without_conda_env_uses_default_env_with_expected_dependencies(
+    auto_arima_object_model,
+):
+    artifact_path = "model"
+    with mlflow.start_run():
+        mlflow.pmdarima.log_model(auto_arima_object_model, artifact_path)
+        model_uri = mlflow.get_artifact_uri(artifact_path)
+    _assert_pip_requirements(model_uri, mlflow.pmdarima.get_default_pip_requirements())
+
+
+def test_pmdarima_pyfunc_serve_and_score(auto_arima_model):
+
+    # NB: dev version of pmdarima uses a __version__ tag of "0.0.0" which will fail the env build.
+    # To be as safe as possible in cross version tests, use the latest released version when
+    # testing dev pmdarima.
+
+    def get_latest_version():
+        ver = pmdarima.__version__
+        if ver == "0.0.0":
+            url = "https://pypi.org/pypi/pmdarima/json"
+            resp = requests.get(url).json()
+            releases = resp["releases"].keys()
+            return sorted(releases, key=parse_version, reverse=True)[0]
+        else:
+            return ver
```

Comment:
Can we use `--no-conda` flag instead?

Reply:
Example:

https://github.com/mlflow/mlflow/blob/b81a1d3ca377ff8ac47d33c9e6111172a31bd145/tests/xgboost/test_xgboost_model_export.py#L35

Reply:
@BenWilson2 Can we address this comment?

Reply:
most definitely. Thank you for the tip @harupy !